js/onoffice-honeypot.js

```diff
@@ -1,23 +1,41 @@
-var form = form || {};
-form.type = form.type || '';
 jQuery(document).ready(function ($) {
-    if (form.type !== 'applicantsearch') {
-        var messageTextarea = $('textarea[name="message"]');
-        var messageInput = $('input[name="message"]');
-        if(messageTextarea.length === 1){
-            messageTextarea.attr('name', 'tmpField');
-        } else if (messageInput.length === 1)  {
-            messageInput.attr('name', 'tmpField');
-        }
-        var label = $('<label>').text('Message:').attr("class", "message");
-        var input = $('<input>').attr({
-            'type': 'text',
-            'name': 'message',
-            'class': 'message'
+    const formElements = getFormsBySpecialHtmlAttribute();
+
+    addHoneypotToForms(formElements);
+
+    function getFormsBySpecialHtmlAttribute() {
+        const formElements = [];
+        const forms = document.querySelectorAll('.oo-form, #onoffice-form');
+
+        forms.forEach(form => {
+            const dataAttr = form.getAttribute('data-applicant-form-id');
+            if (dataAttr === null) {
```

Comment:
I think you are checking here if the form is an ApplicantSearchForm. 
Previously it was checked with "form.type !== 'applicantsearch'". I find that more readable and more stable than the new solution. Can we continue to use checking with form.type or did it not work?

Overall I like the refactoring very much. Good work. :)

Reply:
@andernath If there are multiple forms on the same page, using form.type to check for an applicant search form will cause the honeypot feature to not work on other forms (contact, interest, owner).

Reply:
@yeneastgate what a pity. :/ Thank you for info.